ibis/backends/snowflake/__init__.py

```diff
@@ -472,6 +474,48 @@ def list_schemas(
 
         return self._filter_with_like(schemata, like)
 
+    def list_tables(
+        self,
+        like: str | None = None,
+        database: str | None = None,
+        schema: str | None = None,
+    ) -> list[str]:
+        """List the tables in the database.
+
+        Parameters
+        ----------
+        like
+            A pattern to use for listing tables.
+        database
+            The database (catalog) to perform the list against.
+        schema
+            The schema inside `database` to perform the list against.
+
+            ::: {.callout-warning}
+            ## `schema` refers to database hierarchy
+
+            The `schema` parameter does **not** refer to the column names and
+            types of `table`.
+            :::
+        """
+        query = "SHOW TABLES"
+
+        if database is not None and schema is None:
+            raise com.UnsupportedArgumentError(
+                f"{self.name} cannot list tables only using `database` specifier. "
+                "Include a `schema` argument."
+            )
+        database = (
+            sg.exp.Table(catalog=database, db=schema).sql(dialect=self.name) or None
```

Comment:
Hm, I think we may need to use an API that handles the quoting here, like `sg.table(schema, db=database, quoted=True)`

Reply:
hmm, yeah.  I'll need to also add a little extra handling for passing in a dotted path to `database` to support this being backwards compatible for the moment.  At least all of the extra code will go away in 8.x

Reply:
Hm, I'm not sure we ever supported dotted paths with `database`. Might be okay to not handle that case.

Reply:
ugh -- well, if we use `sg.table` then passing in a dotted path to `schema` will break